src/main/java/org/janelia/saalfeldlab/conversion/CommandLineConverter.java

```diff
@@ -391,25 +389,30 @@ private static void handleRawDataset(
 		writer.setAttribute(dataGroup, "multiScale", true);
 
 		final String outputDataset = Paths.get(dataGroup, "s0").toString();
-		N5ConvertSpark.convert(sc,
-				() -> N5Helpers.n5Reader(inputN5),
-				inputDataset,
-				() -> new N5FSWriter(outputN5, DEFAULT_BUILDER),
-				outputDataset,
-				Optional.of(blockSize),
-				Optional.of(new GzipCompression()), // TODO pass compression
-				// as parameter
-				Optional.ofNullable(null),
-				Optional.ofNullable(null),
-				false);
+
+		if (Paths.get(inputN5).equals(Paths.get(outputN5)) && Paths.get(inputDataset).equals(Paths.get(outputDataset))) {
```

Comment:
Checking for consistency of dataset attributes (dimensions, block size) would be useful here. Resolution and offset will probably be overwritten, so no need to check here.

Reply:
Maybe I'm misunderstanding your suggestion, but in this case the input dataset is also the target, so no checks are necessary.

I thought about checking if the block size of the existing dataset `s0` is the same as requested through the command line, but then figured that it would only make it harder for users because it would require to pass the block size parameter just to make the validation go through (if the block size of the existing dataset is different from the default `64,64,64`).

Resolution and offset will be written into the group attributes under `volumes/raw/data`, so the dataset attributes under `volumes/raw/data/s0` will be left intact. (in fact in my use-case `s0` is a symlink to a read-only directory in another location, which works well)

Reply:
Yes, you are right, brain glitch on my side.